pkg/cluster/actions.go

```diff
@@ -0,0 +1,121 @@
+package cluster
+
+import (
+	"fmt"
+	"github.com/zalando-incubator/postgres-operator/pkg/spec"
+	"github.com/zalando-incubator/postgres-operator/pkg/util"
+	"k8s.io/api/core/v1"
+)
+
+var NoActions []Action = []Action{}
+
+type MetaData struct {
+	cluster   *Cluster
+	namespace string
+}
+
+type CreateSecret struct {
+	ActionSecret
+}
+
+func NewCreateSecret(username string, secret *v1.Secret, cluster *Cluster) CreateSecret {
+	return CreateSecret{ActionSecret{
+		meta: MetaData{
+			cluster: cluster,
+		},
+		secretUsername: username,
+		secret:         secret,
+	}}
+}
+
+func NewUpdateSecret(username string, secret *v1.Secret, cluster *Cluster) UpdateSecret {
+	return UpdateSecret{ActionSecret{
+		meta: MetaData{
+			cluster: cluster,
+		},
+		secretUsername: username,
+		secret:         secret,
+	}}
+}
+
+type UpdateSecret struct {
+	ActionSecret
+}
+
+type ActionSecret struct {
+	meta           MetaData
+	secretUsername string
+	secret         *v1.Secret
+}
+
+type Action interface {
+	Name() string
+	Validate() error
```

Comment:
so the intention is to validate before applying but it is up to the interface user to make so ?